ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/views/imagebrowser/ImageBrowserView.java

```diff
@@ -190,11 +205,15 @@ public void createPartControl(final Composite parent) {
 		typeCombo.addSelectionListener(widgetSelectedAdapter(e -> {
 			mFilters.clear();
 			mFilters.add(textPatternFilter);
-			Combo source = (Combo) e.getSource();
-			switch (source.getSelectionIndex()) {
-				case 0 -> mFilters.add(enabledIcons);
-				case 1 -> mFilters.add(disabledIcons);
-				case 2 -> mFilters.add(wizard);
+			List<String> options = List.of("Enabled", "Disabled", "Wizard", "AllIcons"); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$
```

Comment:
I think the indirection through this list doesn't make the code easier to maintain, as we now have to keep that list in sync with the list of labels defined for the combo above.

I agree it's not ideal to just rely on hard-coded indices, but I think currently we have not really completely clean and simple solution, because the labels are not constants (for the compiler).
Therefore I would leave this part as it is.

Reply:
I agree. I changed the switch to hard indices and changed the condition for adding the SVGFilter to the following (only add SVGs if the selectionIndex is the last possible index):

if (selectionIndex != source.getItemCount() - 1) {
				mFilters.add(svgFilter);
			}